src/Features/LanguageServer/Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs

```diff
@@ -76,14 +77,15 @@ protected override ImmutableArray<Document> GetOrderedDocuments(RequestContext c
             return ImmutableArray.Create(context.Document);
         }
 
-        protected override Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(
+        protected override async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(
             RequestContext context, Document document, Option2<DiagnosticMode> diagnosticMode, CancellationToken cancellationToken)
         {
             // For open documents, directly use the IDiagnosticAnalyzerService.  This will use the actual snapshots
             // we're passing in.  If information is already cached for that snapshot, it will be returned.  Otherwise,
             // it will be computed on demand.  Because it is always accurate as per this snapshot, all spans are correct
             // and do not need to be adjusted.
-            return _analyzerService.GetDiagnosticsAsync(document.Project.Solution, documentId: document.Id, cancellationToken: cancellationToken);
+            return await _analyzerService.GetDiagnosticsForSpanAsync(
+                document, range: null, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
💡 Consider changing this method back to direct pass-through (i.e. remove the `async`/`await` to return the task directly)

Reply:
oh good point.